isobus/src/can_partnered_control_function.cpp

```diff
@@ -205,20 +206,18 @@ namespace isobus
 		return retVal;
 	}
 
-	std::uint32_t PartneredControlFunction::get_number_partnered_control_functions()
+	std::size_t PartneredControlFunction::get_number_partnered_control_functions()
 	{
 		return partneredControlFunctionList.size();
 	}
 
-	ParameterGroupNumberCallbackData PartneredControlFunction::get_parameter_group_number_callback(std::uint32_t index) const
+	std::unique_ptr<ParameterGroupNumberCallbackData> PartneredControlFunction::get_parameter_group_number_callback(std::size_t index) const
 	{
-		ParameterGroupNumberCallbackData retVal(0, nullptr, nullptr);
-
 		if (index < get_number_parameter_group_number_callbacks())
 		{
-			retVal = parameterGroupNumberCallbacks[index];
+			return std::make_unique<ParameterGroupNumberCallbackData>(parameterGroupNumberCallbacks[index]);
```

Comment:
I begin to doubt, doesn't this also make a copy of parameterGroupNumberCallbacks[index]? If so, it might be more logical to return a reference instead I think?

Reply:
This does construct a new object and calls the copy constructor of the underlying object which could be a bit expensive since this is a pretty hot loop. Returning a reference makes sense I think!